mpp/include/mpp_domains_util.inc

```diff
@@ -1730,10 +1792,110 @@ end subroutine mpp_get_tile_compute_domains
      enddo
      domain_out%tile_id = domain_in%tile_id
 
+     if (associated(domain_in%pearray)) then
+         starting = lbound(domain_in%pearray)
+         ending = ubound(domain_in%pearray)
+
+         allocate(domain_out%pearray(starting(1):ending(1), starting(2):ending(2)))
```

Comment:
This just looks like it's ready to leak memory

Reply:
Is it possible to have the unit test run a pathological case to test for leakage?

Reply:
It is possible to leak memory if it not used correctly (even without this code change). 

For example, 
A user can do:
```F90
call mpp_copy_domain(Domain, Domain2)
call mpp_copy_domain(Domain, Domain2)
```

This will "work", but it will leak memory. 

Not sure what to do about this ...

Reply:
Shouldn't we have a check for domain_out to see if it has been previously used?  If so, I can see a runtime error when domain_out%pearray is allocated.

Reply:
I added this to the top of the subroutine. Now the code crashes if you try to call mpp_copy_domain twice. 
```F90  
   if (associated(domain_out%x)) then
         call mpp_error(FATAL, "mpp_copy_domain: domain_out is already set")
     endif

```